src/sensor_module.c

```diff
@@ -0,0 +1,232 @@
+ * Global Data
+ ******************************************************************************/
+
+/******************************************************************************
+ * Module Static Data
+ ******************************************************************************/
+
+/******************************************************************************
+ * Local Function Prototypes
+ ******************************************************************************/
+
+/******************************************************************************
+ * Function Definitions
+ ******************************************************************************/
+/**
+ * Reads the compass value in the file "dataStore.bin"
+ *
+ * @return     returns the compass value
+ */
+int16_t readCompass(void)
+{
+    Compass_Sim_Config_t simConfig;
+    Compass_Config_t compassConfig;
+    int16_t compassValue;
+    simConfig.path = "dataStore.bin";
+    compassConfig.measMode = Compass_Mode_Continuous;
+
+    Compass_Sim_Init(&simConfig);
+    Compass_Init(&compassConfig);
+    Compass_Read(&compassValue);
+    Compass_Sim_Deinit();
+    return compassValue;
+}
+
+/**
+ * Reads the voltage value in the file "voltageSim.bin"
+ *
+ * @return     returns the voltage value
+ */
+int16_t readVoltage(void)
+{
+    Voltage_Sim_Config_t simConfig;
+    uint16_t voltageValue;
+    simConfig.path = "voltageSim.bin";
+
+    Voltage_Sim_Init(&simConfig);
+    Voltage_Init();
+    Voltage_Read(&voltageValue);
+    Voltage_Sim_Deinit();
+    return voltageValue;
+}
+
+/**
+ * Changes the time from year, month, day, hours, minutes, seconds, to time since Unix Epoch
+ *
+ * @param      year   The year
+ * @param      month  The month
+ * @param      day    The day
+ * @param      hms    The hours, minutes, and seconds
+ *
+ * @return     returns the time since Unix Epoch
+ */
+uint64_t changeTime(int *year, uint8_t *month, uint8_t *day, uint8_t hms[3])
+{
+    struct tm t;
+    time_t t_of_day;
+
+    t.tm_year = *year - 1900;      // Year - 1900
+    t.tm_mon = *month - 1;           // Month, where 0 = january
+    t.tm_mday = *day;            // Day of the month
+    t.tm_hour = hms[0];
+    t.tm_min = hms[1];
+    t.tm_sec = hms[2];
+    t.tm_isdst = -1;            // Is DST on? 1 = yes, 0 = no, -1 = unknown
+    t_of_day = mktime(&t);
+
+    return t_of_day;
+}
+
+/**
+ * Sets the constant portion of the DataSensorPacket struct
+ *
+ * @param      packet  The packet
+ */
+void setPacket(DataSensorPacket_t *packet)
+{
+    packet->sync_char1 = 0xE4;
+    packet->sync_char2 = 0xEB;
+    packet->packet_class = 0x04;
+    packet->packet_id = 0x03;
+    packet->payload.version = 0x01;
+    packet->payload_length = sizeof(DataSensorInput_t);
+    packet->payload.heading = readCompass(); //the number in dataStore.bin (pValue);
+    packet->payload.voltage = readVoltage(); //the number in voltageSim.bin (pValue);
```

Comment:
These are not constants - the docstring says this function sets the constant portion...